src/main/java/net/snowflake/client/jdbc/SnowflakeSQLException.java

```diff
@@ -100,12 +112,23 @@ public SnowflakeSQLException(Throwable ex, ErrorCode errorCode, Object... params
     this(ex, errorCode.getSqlState(), errorCode.getMessageCode(), params);
   }
 
+  /**
+   * @deprecated use {@link SnowflakeSQLException#SnowflakeSQLException(String, Throwable, String,
+   *     int, Object...)}
+   */
+  @Deprecated
```

Comment:
Please check with Kiran/Anurag for deprecation process. Even we may need to update the doc, if we doc this API.

Reply:
I checked in the docs that:
- SnowflakeConnection methods mentioned in https://docs.snowflake.com/en/developer-guide/jdbc/jdbc-using#upload-data-files-directly-from-a-stream-to-an-internal-stage and https://docs.snowflake.com/en/developer-guide/jdbc/jdbc-using#download-data-files-directly-from-an-internal-stage-to-a-stream weren't changed
- Description in API Support does not include changed methods https://docs.snowflake.com/en/developer-guide/jdbc/jdbc-api

Reply:
We discussed internally that those public classes are not part of public documentation, javadoc is updated and it's only the thing how to reflect those changes in release notes - so we can merge this PR